src/test/java/com/microsoft/graph/requests/extensions/GraphServiceClientTest.java

```diff
@@ -0,0 +1,211 @@
+package com.microsoft.graph.requests.extensions;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+
+import com.microsoft.graph.authentication.IAuthenticationProvider;
+import com.microsoft.graph.concurrency.DefaultExecutors;
+import com.microsoft.graph.concurrency.ICallback;
+import com.microsoft.graph.concurrency.IExecutors;
+import com.microsoft.graph.concurrency.IProgressCallback;
+import com.microsoft.graph.core.ClientException;
+import com.microsoft.graph.core.DefaultClientConfig;
+import com.microsoft.graph.http.DefaultHttpProvider;
+import com.microsoft.graph.http.IHttpProvider;
+import com.microsoft.graph.http.IHttpRequest;
+import com.microsoft.graph.http.IStatefulResponseHandler;
+import com.microsoft.graph.logger.ILogger;
+import com.microsoft.graph.logger.LoggerLevel;
+import com.microsoft.graph.models.extensions.IGraphServiceClient;
+import com.microsoft.graph.serializer.DefaultSerializer;
+import com.microsoft.graph.serializer.ISerializer;
+
+public class GraphServiceClientTest {
+
+    @Test
+    public void testClientMethodsReturnStuff() {
+        IGraphServiceClient client = GraphServiceClient.fromConfig(new DefaultClientConfig() {
```

Comment:
These tests fail because AuthenticationProvider cannot be null. You can pass in an instance of `MockAuthenticationProvider` for testing purposes and `assertNotNull`

Reply:
Ah some validation turned up that didn't exist before. The authentication provider must always be provided. I'm going to adjust the builder path to make this happen but also I suggest that we make the DefaultConfig.getAuthenticationProvider method abstract to force implementation. I'll update the PR and see what you think.